Principal surprise

lincmba opened the pull request:
Principal surprise code and tests
Poetry to manage package installation
CircleCI set up

lincmba committed:
Principal surprise code and tests.
 Poetry to manage package installation

lincmba committed:
Add circleci config

gstuder-ona commented on pyproject.toml:
```diff
@@ -0,0 +1,31 @@
+shapely = "^1.7.0"
+folium = "^0.10.1"
+scikit-learn = "^0.22.2"
+mlrose = "^1.3.0"
```
don't need mlrose

gstuder-ona commented on pyproject.toml:
```diff
@@ -0,0 +1,31 @@
+[tool.poetry.dependencies]
+python = "^3.7"
+plac = "^1.1"
+geojson = "^2.5.0"
```
don't need geojson, shapely, folium

gstuder-ona commented on pyproject.toml:
```diff
@@ -0,0 +1,31 @@
+scikit-learn = "^0.22.2"
+mlrose = "^1.3.0"
+jsonpickle = "^1.3"
+aiohttp-index = "^0.1"
```
dont' need aiohttp stuff

gstuder-ona commented on pyproject.toml:
```diff
@@ -0,0 +1,31 @@
+dask = "^2.26.0"
+numpy = "^1.19.2"
+toolz = "^0.10.0"
+tornado = "^6.0.4"
```
don't think we need tornado / fsspec?

lincmba replied:
Installing dask using `poetry add dask` not all modules are installed, as mentioned here https://docs.dask.org/en/latest/install.html#pip . Tests fail with `ImportError: Dask dataframe requirements are not installed.` until `fsspec` and`fsspec` are installed

lincmba replied:
Installing Dask using `poetry add dask` misses some modules https://docs.dask.org/en/latest/install.html#pip. Tests then fail with `ImportError: Dask dataframe requirements are not installed.` until `toolz` and `fsspec` are installed

gstuder-ona commented on pyproject.toml:
```diff
@@ -0,0 +1,31 @@
+aiohttp-index = "^0.1"
+matplotlib = "^3.2.1"
+colorhash = "^1.0.2"
+jsonpath-ng = "^1.5.1"
```
not sure we need jsonpath-ng either

gstuder-ona commented on pyproject.toml:
```diff
@@ -0,0 +1,31 @@
+black = { version = "*", allow-prereleases = true }
+pytest = "^5.3"
+flake8 = "^3.7"
+pytest-asyncio = "^0.10.0"
```
shouldn't need the pytest-asyncio but maybe useful later?

gstuder-ona commented on perf_tools/principal_surprise.py:
```diff
@@ -0,0 +1,835 @@
+from collections import namedtuple
+
+
+class AbnormalTaskPerfAnalyzer:
```
This is fine for first pass but suspect we probably want a second patch renaming this and importing it via __init__.py

pld replied:
I think I'd rather see this class in its own file with a file name matching the class name but as snack case

gstuder-ona commented on .circleci/config.yml:
```diff
@@ -0,0 +1,69 @@
+          name: run tests
+          command: poetry run pytest
+          environment:
+            REMOTE_DOCKER_SSH_HOST: remote-docker
```
shouldn't need this variable

gstuder-ona reviewed (changes requested):
Can we add one more patch to kill the unneeded dependencies?  CircleCI should validate that things work fine afterwards.

pld reviewed (commented):
can we change the base folder from `perf_tools` to... `chti`? `community-health-task-investigator`

lincmba committed:
Rename package
Remove unnecessary circleci environment variable
Remove dependencies we dont need

gstuder-ona commented on pyproject.toml:
```diff
@@ -0,0 +1,23 @@
+colorhash = "^1.0.2"
+pandas = "^1.1.2"
+numpy = "^1.19.2"
+dask = "^2.26.0"
```
yeah the line here is:
dask = { version = "^2.9", extras = ["dataframe"]}

you just need to add the extras, like in pip. If you edit the file and then `poetry update` you should get the packages you need.

gstuder-ona commented on pyproject.toml:
```diff
@@ -0,0 +1,23 @@
+python = "^3.7"
+plac = "^1.1"
+scikit-learn = "^0.22.2"
+jsonpickle = "^1.3"
```
don't need jsonpickle

gstuder-ona commented on pyproject.toml:
```diff
@@ -0,0 +1,23 @@
+numpy = "^1.19.2"
+dask = "^2.26.0"
+toolz = "^0.10.0"
+fsspec = "^0.8.2"
```
don't need fsspec

moshthepitt commented:
:)

Do we want to set up [pre-commit](https://pre-commit.com/) with black, isort, flake8, and optionally pylint and mypy? ([xlson example](https://github.com/OpenSRP/xlson)).

I know its a nitpicky comment, but I think getting "strong/opinionated" linting and formatting assistance makes for a healthier and more standardized developer experience in the long run.

gstuder-ona commented:
> :)
> 
> Do we want to set up [pre-commit](https://pre-commit.com/) with black, isort, flake8, and optionally pylint and mypy? ([xlson example](https://github.com/OpenSRP/xlson)).
> 
> I know its a nitpicky comment, but I think getting "strong/opinionated" linting and formatting assistance makes for a healthier and more standardized developer experience in the long run.

Agree - not sure if it belongs in this patch vs a second patch but that's definitely why we pull in the flake8/black tools as a dependency.  Maybe we create a ticket for this to discuss?

It can't be annoying though - I find hooking into the git commits themselves kind of the wrong place, 1) because it interferes with my local development and rebasing and 2) because it requires an extra setup step in a new repo.  I kind of like the one-liner `poetry run autostyle` as implemented here: https://github.com/onaio/canopy-factuate/blob/master/pyproject.toml#L46 

Happy to discuss more, I haven't tried pre-commit specifically: https://github.com/onaio/canopy/issues/1494

pld commented:
Agree let’s add it, let’s not block this b/c of it

> On Sep 18, 2020, at 9:12 AM, Greg Studer <notifications@github.com> wrote:
> 
> 
> :)
> 
> Do we want to set up pre-commit <https://pre-commit.com/> with black, isort, flake8, and optionally pylint and mypy? (xlson example <https://github.com/OpenSRP/xlson>).
> 
> I know its a nitpicky comment, but I think getting "strong/opinionated" linting and formatting assistance makes for a healthier and more standardized developer experience in the long run.
> 
> Agree - not sure if it belongs in this patch vs a second patch but that's definitely why we pull in the flake8/black tools as a dependency. Maybe we create a ticket for this to discuss?
> 
> It can't be annoying though - I find hooking into the git commits themselves kind of the wrong place, 1) because it interferes with my local development and rebasing and 2) because it requires an extra setup step in a new repo. I kind of like the one-liner poetry run autostyle as implemented here: https://github.com/onaio/canopy-factuate/blob/master/pyproject.toml#L46 <https://github.com/onaio/canopy-factuate/blob/master/pyproject.toml#L46>
> Happy to discuss more, I haven't tried pre-commit specifically: onaio/canopy#1494 <https://github.com/onaio/canopy/issues/1494>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub <https://github.com/onaio/community-health-task-investigator/pull/2#issuecomment-694859490>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AAAMMERVAZPYFN5UTRVROTTSGNMETANCNFSM4RQODSLA>.
>

gstuder-ona committed:
Fix poetry dependencies

gstuder-ona commented:
@lincmba I have a patch up here: https://github.com/onaio/community-health-task-investigator/pull/3/commits/1fecc58e2a77ce011c9379fb49d395304d0cbe2e that has what I think are the minimal pyproject.toml requirements (and some stuff that's helpful for pyenv, if you use that).  If you haven't figured it out already, the tests all run etc. with these settings.

Don't worry about merge conflicts or anything either, I can easily just rebase after your patch is done.  Renaming stuff is always finicky.

gstuder-ona commented on pyproject.toml:
```diff
@@ -0,0 +1,23 @@
+name = "chti"
+version = "0.1.0"
+description = ""
+authors = ["Sébastien Eustace <sebastien@eustace.io>"]
```
Oh - do we definitely need to change this @lincmba?  Would be embarrassing to share :-)

lincmba replied:
Missed that. Fixing it

gstuder-ona reviewed (changes requested):
Once @pld is happy I'm happy, my only nitpick is that we should definitely change the author.  devs@ona.io?
